### PR TITLE
F implement create campaign endpoint for givethio projects

### PR DESCRIPTION
**Pull request by mohammadranjbarz**
related to Giveth/giveth-dapp#2436

### CODE REVIEW

File: src/services/createCampaignForGivethioProjects/createCampaignForGivethioProjects.service.js [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+      const { txHash, image, slug } = data;
+
+      const projectInfo = await givethIoAdapter.getProjectInfoBySLug(slug);
+      const {
+        id: givethIoProjectId,
+        title,
+        description,
+        walletAddress: ownerAddress,
+      } = projectInfo;
+      if (params.user.address !== ownerAddress) {
```
Comment (aminlatifi) [inline src/services/createCampaignForGivethioProjects/createCampaignForGivethioProjects.service.js:22]: We don't know ownerAddress in giveth.io backend has correct checksum
```suggestion
      if (params.user.address.toLowerCase() !== ownerAddress.toLowerCase()) {
```

File: src/services/createCampaignForGivethioProjects/createCampaignForGivethioProjects.service.js [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+        throw new errors.Forbidden('The owner of project in givethIo is not you');
+      }
+      let campaign = await findCampaignByGivethIoProjectId(app, givethIoProjectId);
+      if (campaign) {
+        throw new errors.BadRequest('Campaign with this givethIo projectId exists');
+      }
+      campaign = await app.service('campaigns').create({
```
Comment (aminlatifi) [inline src/services/createCampaignForGivethioProjects/createCampaignForGivethioProjects.service.js:29]: Will campaign service allow creating a campaign with this ownerAddress? (ownerAddress doesn't have project owner role)

**Review by aminlatifi — CHANGES_REQUESTED** [review]

File: src/app.hooks.js [hunk 3]
```diff
@@ -27,6 +27,10 @@ const authenticate = () => context => {
     // for creating donations it's not needed to be authenticated, anonymous users can donate
     return context;
   }
+  if (context.path === 'createCampaignForGivethioProjects' && context.method === 'create') {
```
[the remark this answers is not in the document]
Comment (mohammadranjbarz) [reply src/app.hooks.js]: You are right, it seems good , done

File: src/services/createCampaignForGivethioProjects/createCampaignForGivethioProjects.service.js [hunk 4]
```diff
@@ -0,0 +1,99 @@
+const errors = require('@feathersjs/errors');
+const config = require('config');
+const logger = require('winston');
+const { findCampaignByGivethIoProjectId } = require('../../repositories/campaignRepository');
+const { getGivethIoAdapter } = require('../../adapters/adapterFactory');
+
+const givethIoAdapter = getGivethIoAdapter();
+module.exports = function aggregateDonations() {
```
[the remark this answers is not in the document]
Comment (mohammadranjbarz) [reply src/services/createCampaignForGivethioProjects/createCampaignForGivethioProjects.service.js]: done

File: src/services/createCampaignForGivethioProjects/createCampaignForGivethioProjects.service.js (hunk 1 above)
Comment (mohammadranjbarz) [reply src/services/createCampaignForGivethioProjects/createCampaignForGivethioProjects.service.js]: Good point,
done

**Comment by aminlatifi** [comment]
LGTM

**Pull request merged**